src/cpu/x64/jit_uni_planar_convolution.cpp

```diff
@@ -0,0 +1,171 @@
+/*******************************************************************************
+* Copyright 2019-2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "jit_uni_planar_convolution.hpp"
+
+#include "common/c_types_map.hpp"
+#include "common/dnnl_thread.hpp"
+#include "common/memory_tracking.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace x64 {
+
+using namespace dnnl::impl::status;
+using namespace dnnl::impl::utils;
+
+#define src_blk_off(f, n, c, d, h, w) \
+    pd()->ndims() == 5 \
+        ? (f).blk_off(n, c, d, h, w) \
+        : (f).blk_off(n, c, h, w)
+
+#define wht_blk_off(f, g, oc, ic, kd, kh, kw) \
+    pd()->ndims() == 5 \
+        ? pd()->with_groups() \
+            ? (f).blk_off(g, oc, ic, kd, kh, kw) \
+            : (f).blk_off(oc, ic, kd, kh, kw) \
+        : pd()->with_groups() \
+            ? (f).blk_off(g, oc, ic, kh, kw) \
+            : (f).blk_off(oc, ic, kh, kw)
+
+template <cpu_isa_t isa>
+void _jit_uni_planar_convolution_fwd_t<isa>::execute_forward(const exec_ctx_t &ctx) const {
+    auto src = CTX_IN_MEM(const data_t *, DNNL_ARG_SRC);
+    auto weights = CTX_IN_MEM(const data_t *, DNNL_ARG_WEIGHTS);
+    auto bias = CTX_IN_MEM(const data_t *, DNNL_ARG_BIAS);
+    auto dst = CTX_OUT_MEM(data_t *, DNNL_ARG_DST);
+
+    auto MB = CTX_IN_BATCH(DNNL_ARG_SRC);
+
+    const memory_desc_wrapper src_d(pd()->src_md());
+    const memory_desc_wrapper dst_d(pd()->dst_md());
+    const memory_desc_wrapper weights_d(pd()->weights_md(0));
+    const memory_desc_wrapper bias_d(pd()->weights_md(1));
+
+    const auto &jcp = pd()->jcp_;
+
+    int od_indexes[jcp.od];
```

Comment:
There is a error C2131: expression did not evaluate to a constant. I have used MSVC 2019 compiler.

Reply:
thanks! fixed